.github/workflows/build-lint-test.yml

```diff
@@ -52,7 +52,7 @@ jobs:
           name: ${{ inputs.artifact-name }}
       - name: Codecov
         if: ${{ inputs.skip-codecov == false }}
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@894ff025c7b54547a9a2a1e9f228beae737ad3c2
```

Comment:
1. We intentionally don't update this for any new release of Codecov, since the goal of this pinning is to decrease the risk of being affected by malicious updates of third-party actions. If at some point we decide that we want to use a newer version, this update has to be perfomed manually by updating the commit hash.

2. We should also update the third-party actions and tools that we use in other workflows. The ones I found are:
    - Codecov 1.5.2 in https://github.com/cloudscape-design/collection-hooks/blob/main/.github/workflows/build.yml#L28
    - Codecov 3 in https://github.com/cloudscape-design/actions/blob/main/workflows/build-lint-test.yml#L55
    - `conventional-changelog-cli` in https://github.com/cloudscape-design/actions/blob/main/workflows/release-gh-notes.yml#L37 and https://github.com/cloudscape-design/.github/blob/main/.github/workflows/release-gh-notes.yml#L37

The other actions that we use are provided either by Amazon or GitHub and thus trusted, i.e. pinning them is less of a priority.